### PR TITLE
✨ Switch to stylelint

### PR DESCRIPTION
**Pull request by stormwarning**
Removes usage of `scss-lint` and replaces with `stylelint`.

### CODE REVIEW

File: .stylelintrc [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    ],
+    "declaration-empty-line-before": [
+      "always",
+      {
+        "except": ["first-nested"],
+        "ignore": ["after-declaration"]
+      }
+    ],
+
+    "order/order": [
```
Comment (elidupuis) [inline .stylelintrc]: Are these settings trying to mimic our current sorting configuration? Seems like a lot of configuration...

File: .stylelintrc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        },
+        "rules"
+      ]
+    ],
+
+    "plugin/selector-bem-pattern": {
+      "preset": "bem"
+    },
+
+    "scss/at-extend-no-missing-placeholder": true,
```
Comment (elidupuis) [inline .stylelintrc]: I can't tell if these are defaults or overrides; can we just extend `stylelint-scss` and get rid of most/all of these?

Comment (stormwarning) [reply .stylelintrc]: `stylelint-scss` is a plugin, not a configuration to extend—it just provides the extra Sass-specific rules.

File: .stylelintrc (hunk 1 above)
Comment (stormwarning) [reply .stylelintrc]: So these are just setting the order of different elements of a block. So for example (line spaced for clarity)
```css
.selector {
  --custom: 'property';

  $sass: 'variable';

  @extend %placeholder-class;

  @include mixin();

  display: block;
  content: "smacss order";

  @supports (display: grid) {
    display: grid;
  }

  @include content-mixin() {
    max-width: 320px;
  }

  &:hover {}

  &::after {}

  .nested-selector {}
}
```

This sort of order has always made the most sense to me. If you don't think it's worth enforcing I can drop the rules.

Comment (stormwarning) [reply .stylelintrc]: Dropped 'em anyway after I opened a PR on the main config to include them. 😉

File: .stylelintrc (hunk 2 above)
Comment (stormwarning) [reply .stylelintrc]: Extended a different config that includes the Sass-specific rules for us.

File: tests/dummy/app/styles/app.css [hunk 3]
```diff
@@ -0,0 +1 @@
+body { background-color: #caf0ff; }
```
Comment (elidupuis) [inline tests/dummy/app/styles/app.css]: Is this just to have some test CSS in the dummy app?

Comment (stormwarning) [reply tests/dummy/app/styles/app.css]: Yeah, IIRC `stylelint` complained that the CSS file was empty.

File: blueprints/showbie-tooling/index.js [hunk 4]
```diff
@@ -10,18 +11,24 @@ module.exports = {
...
         { name: 'eslint-plugin-ember-suave', target: '~1.0.0' },
-        { name: 'lodash.samplesize', target: '^4.2.0' }
+        { name: 'lodash.samplesize', target: '^4.2.0' },
+        { name: 'stylelint', target: '~8.2.0' },
+        {
+          name: 'stylelint-config-property-sort-order-smacss',
+          target: '~2.0.0'
+        },
+        { name: 'stylelint-config-sass-guidelines', target: '~3.1.0' },
+        { name: 'stylelint-order', target: '^0.7.0' },
```
Comment (elidupuis) [inline blueprints/showbie-tooling/index.js]: Might as well lock with with `~` like the rest of the deps. `^` is the devil.

Comment (stormwarning) [reply blueprints/showbie-tooling/index.js]: I believe in versions without a major version number, `^` acts like `~`. [[reference](https://docs.npmjs.com/misc/semver#caret-ranges-123-025-004)]

Comment (elidupuis) [reply blueprints/showbie-tooling/index.js]: Fair enough. I'm just thinking more about consistency. And what happens when 1.0.0 comes out and someone runs `yarn upgrade stylelint-order`? Does it bump the major version?

**Comment by elidupuis** [comment]
@stormwarning I tried to resolve the conflict here (via GH UI) but it didn't work. 

Also, you forgot to bump the stylelint-config in the blueprint in 602d88828ff4c7af3d9d0919348d2c1e8ab8ed18.

**Pull request merged**